pennylane/transforms/zx/optimize_t_count.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2018-2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains a transform to apply the
+`full_optimize <https://pyzx.readthedocs.io/en/latest/api.html#pyzx.optimize.full_optimize>`__
+pass (available through the external `pyzx <https://pyzx.readthedocs.io/en/latest/index.html>`__ package)
+to a PennyLane Clifford + T circuit.
+"""
+
+from pennylane.tape import QuantumScript, QuantumScriptBatch
+from pennylane.transforms import transform
+from pennylane.typing import PostprocessingFn
+
+from .converter import from_zx, to_zx
+from .helper import _needs_pyzx
+
+
+@_needs_pyzx
+@transform
+def optimize_t_count(tape: QuantumScript) -> tuple[QuantumScriptBatch, PostprocessingFn]:
+    """Reduce the number of T gates in a Clifford + T circuit by using some basic commutation and
+    cancellation rules, combined with a dedicated phase-polynomial optimization strategy based on the
+    `Third Order Duplicate and Destroy (TODD) <https://arxiv.org/abs/1712.01557>`__ algorithm.
+
+    This transform applies a sequence of passes for T-count optimization to the given Clifford + T circuit.
+    First, some ZX-based commutation and cancellation rules are applied to simplify the circuit.
+    Then, the circuit is cut into phase-polynomial blocks and the TODD algorithm is used to optimize each of these phase polynomials.
+    Depending on the number of qubits and T gates in the original circuit, it might take a long time to run.
```

Comment:
Maybe something like this?
```suggestion
    For circuits with many qubits and T gates, this transform may exhibit long run-times.
```